packages/new-editor/src/components/editor/variables/detail/Variable.tsx

```diff
@@ -1,50 +1,45 @@
-import { Fieldset, Input, Message, SimpleSelect, Textarea } from '@axonivy/ui-components';
+import { Fieldset, Input, Message, Textarea } from '@axonivy/ui-components';
 import { getNode, updateNode, hasChildren as variableHasChildren } from '../../../../data/data';
-import type { TreePath } from '../../../../types/config';
-import { metadataOptions, type Metadata, type Variable } from '../../data/Variable';
+import { treeNodeNameAttribute, type TreePath } from '../../../../types/config';
+import { variableDescriptionAttribute, type Variable, type VariableUpdates } from '../../data/Variable';
+import { Metadata } from './Metadata';
+import { Value } from './Value';
 
 type VariableProps = {
   variables: Array<Variable>;
   variablePath: TreePath;
   setVariables: (variables: Array<Variable>) => void;
 };
 
-export const VariableDetail = ({ variables, variablePath, setVariables }: VariableProps) => {
+export const VariablesDetail = ({ variables, variablePath, setVariables }: VariableProps) => {
   const variable = getNode(variables, variablePath);
   if (!variable) {
     return <Message>Select a variable to edit.</Message>;
   }
 
   const hasChildren = variableHasChildren(variable);
 
-  const handleVariableAttributeChange = <TKey extends keyof Variable>(key: TKey, value: Variable[TKey]) => {
-    const newVariables = structuredClone(variables);
-    updateNode(newVariables, variablePath, key, value);
+  const handleVariableAttributeChange = (updates: VariableUpdates) => {
+    const newVariables = updateNode(variables, variablePath, updates);
     setVariables(newVariables);
```

Comment:
be careful that you don't accidentally modify the state inside of updateNode! https://react.dev/reference/rules/components-and-hooks-must-be-pure#state

better use setVariable(old => updateNode(old, varialbePath, updates));